compiler+runtime/test/cpp/jank/runtime/obj/persistent_vector.cpp

```diff
@@ -0,0 +1,73 @@
+#include <jank/runtime/obj/character.hpp>
+#include <jank/runtime/obj/number.hpp>
+#include <jank/runtime/obj/nil.hpp>
+#include <jank/runtime/core.hpp>
+#include <jank/runtime/context.hpp>
+
+/* This must go last; doctest and glog both define CHECK and family. */
+#include <doctest/doctest.h>
+
+namespace jank::runtime::obj
+{
+  TEST_SUITE("persistent_vector")
+  {
+    static persistent_vector_ptr const v(persistent_vector::empty()
+                                           ->conj(make_box('f'))
+                                           ->conj(make_box('o'))
+                                           ->conj(make_box('o'))
+                                           ->conj(make_box(' '))
+                                           ->conj(make_box('b'))
+                                           ->conj(make_box('a'))
+                                           ->conj(make_box('r')));
```

Comment:
1. We can use `auto`
2. We can use `make_box` and the `in_place` ctor to provide all values at once

```cpp
  static auto const v{
    make_box<persistent_vector>(std::in_place,
      make_box('f'),
      make_box('o'),
      make_box('o'),
      make_box(' '),
      make_box('b'),
      make_box('a'),
      make_box('r') };
```